app/src/main/java/org/xbmc/kore/service/library/SyncMusic.java

```diff
@@ -151,6 +146,8 @@ private void deleteMusicInfo(final ContentResolver contentResolver,
                                where, new String[]{String.valueOf(hostId)});
         contentResolver.delete(MediaContract.AlbumGenres.CONTENT_URI,
                                where, new String[]{String.valueOf(hostId)});
+        contentResolver.delete(MediaContract.SongArtists.CONTENT_URI,
+                               where, new String[]{String.valueOf(hostId)});
```

Comment:
Argh, yes this should have been in the original PR...
Fortunately, the table song_artists requires the combination host_id, songid, and artistid to be unique. 
( UNIQUE (host_id, songid, artistid) ). This means, during additional syncs the entries will simply replace the current entries.